File: Chapter 06/Java/src/main/java/com/grokkingsoftwarearchitecture/chapter06/section_6_3_2_graphql/FakeGraphQLHandler.java
package com.grokkingsoftwarearchitecture.chapter06.section_6_3_2_graphql;
/**
 * THE FAKE ENDPOINT.
 * Intercepts the HTTP call and returns our precise GraphQL JSON response.
 */
public class FakeGraphQLHandler {
    
    /**
     * Simulates an HTTP POST request to a remote GraphQL server.
     *
     * @param url The mock endpoint being called.
     * @param payload The JSON-encoded GraphQL query.
     * @return A raw JSON string containing exactly what was asked for.
     */
    public String post(String url, String payload) {
        // The exact JSON response. Notice there is NO over-fetching here!
        return "{\n" +
               "  \"data\": {\n" +
               "    \"chipItem\": { \"name\": \"Salt & Vinegar Chips\" },\n" +
               "    \"sodaItem\": { \"price\": 1.50 }\n" +
               "  }\n" +
               "}";
    }
}

File: Chapter 06/C#/Section_6_3_2_GraphQL/FakeGraphQLHandler.cs
using System.Net;

namespace Chapter06.GraphQLExample
{
    // THE FAKE ENDPOINT
    // Intercepts the HTTP call and returns our precise GraphQL JSON response.
    public class FakeGraphQLHandler : HttpMessageHandler
    {
        protected override Task<HttpResponseMessage> SendAsync(
            HttpRequestMessage request,
            CancellationToken cancellationToken)
        {
            // The exact JSON response from Listing 6.2
            // Notice there is NO over-fetching here!
            string jsonResponse =
            @"{
                ""data"": {
                    ""chipItem"": { ""name"": ""Salt & Vinegar Chips"" },
                    ""sodaItem"": { ""price"": 1.50 }
                }
             }";
            // Package it up as a standard HTTP 200 OK response
            var response = new HttpResponseMessage(HttpStatusCode.OK)
            {
                Content = new StringContent(
                    jsonResponse,
                    System.Text.Encoding.UTF8,
                    "application/json")
            };

            return Task.FromResult(response);
        }
    }
}
